pkg/build/rpm.go

```diff
@@ -1,60 +1,116 @@
+
+//go:embed scripts/rpms/10-rpm-install.sh.tpl
+var modifyRPMScript string
 
-	rpms, err := os.ReadDir(rpmSourceDir)
+func (b *Builder) processRPMs() error {
+	RPMSourceDir, err := b.generateRPMPath()
 	if err != nil {
-		return nil, fmt.Errorf("reading rpm source dir: %w", err)
+		return fmt.Errorf("generating RPM path: %w", err)
+	}
+	// Only proceed with processing the RPMs if the directory exists
+	if RPMSourceDir == "" {
+		return nil
 	}
 
-	for _, rpmFile := range rpms {
-		if filepath.Ext(rpmFile.Name()) == ".rpm" {
-			rpmFileNames = append(rpmFileNames, rpmFile.Name())
-		}
+	RPMFileNames, err := getRPMFileNames(RPMSourceDir)
+	if err != nil {
+		return fmt.Errorf("getting RPM file names: %w", err)
+	}
+
+	err = copyRPMs(RPMSourceDir, b.combustionDir, RPMFileNames)
+	if err != nil {
+		return fmt.Errorf("copying RPMs over: %w", err)
 	}
 
-	if len(rpmFileNames) == 0 {
-		return nil, fmt.Errorf("no rpms found")
+	err = b.writeRPMScript(RPMFileNames)
+	if err != nil {
+		return fmt.Errorf("writing the RPM install script %s: %w", modifyRPMScriptName, err)
 	}
 
-	return rpmFileNames, nil
+	return nil
 }
 
-func (b *Builder) copyRPMs() error {
-	rpmSourceDir := filepath.Join(b.buildConfig.ImageConfigDir, "rpms")
-	// Only proceed with copying the RPMs if the directory exists
-	_, err := os.Stat(rpmSourceDir)
+func getRPMFileNames(RPMSourceDir string) ([]string, error) {
+	var RPMFileNames []string
+
+	RPMs, err := os.ReadDir(RPMSourceDir)
 	if err != nil {
-		if os.IsNotExist(err) {
-			return nil
+		return nil, fmt.Errorf("reading RPM source dir: %w", err)
+	}
+
+	for _, RPMFile := range RPMs {
+		if filepath.Ext(RPMFile.Name()) == ".rpm" {
+			RPMFileNames = append(RPMFileNames, RPMFile.Name())
 		}
-		return fmt.Errorf("checking for rpm directory at %s: %w", rpmSourceDir, err)
 	}
-	rpmDestDir := b.combustionDir
 
-	rpmFileNames, err := b.getRPMFileNames(rpmSourceDir)
-	if err != nil {
-		return fmt.Errorf("getting rpm file names: %w", err)
+	if len(RPMFileNames) == 0 {
+		return nil, fmt.Errorf("no RPMs found")
 	}
 
-	for _, rpm := range rpmFileNames {
-		sourcePath := filepath.Join(rpmSourceDir, rpm)
-		destPath := filepath.Join(rpmDestDir, rpm)
+	return RPMFileNames, nil
+}
+
+func copyRPMs(RPMSourceDir string, RPMDestDir string, RPMFileNames []string) error {
+	for _, RPM := range RPMFileNames {
+		sourcePath := filepath.Join(RPMSourceDir, RPM)
+		destPath := filepath.Join(RPMDestDir, RPM)
 
-		err = fileio.CopyFile(sourcePath, destPath)
+		err := fileio.CopyFile(sourcePath, destPath)
 		if err != nil {
 			return fmt.Errorf("copying file %s: %w", sourcePath, err)
 		}
 	}
 
 	return nil
 }
+
+func (b *Builder) writeRPMScript(RPMFileNames []string) error {
+	values := struct {
+		RPMs string
+	}{
+		RPMs: strings.Join(RPMFileNames, " "),
+	}
+
+	writtenFilename, err := b.writeCombustionFile(modifyRPMScriptName, modifyRPMScript, &values)
+	if err != nil {
+		return fmt.Errorf("writing RPM script %s: %w", modifyRPMScriptName, err)
```

Comment:
The caller already formats in the name of the script. We can simplify this and reduce the duplication:
```suggestion
		return fmt.Errorf("writing RPM script: %w", err)
```